agent.go

```diff
@@ -117,6 +117,7 @@ type container struct {
 	container libcontainer.Container
 	config    configs.Config
 	processes map[string]*process
+	pod       *pod
```

Comment:
Refactoring sounds okay to me, but you should give the pod pointer through the parameters of the function `closeProcessStreams()` since you need the pod pointer only in this function. This would avoid to add this pointer in every container.

Moreover, please fix your commit message, there are some syntax/grammar errors.

Reply:
This comment has not been addressed.